src/btree/bt_sync.c

```diff
@@ -299,21 +299,9 @@ __wt_sync_file(WT_SESSION_IMPL *session, WT_CACHE_OP syncop)
              * cache clean but with history that cannot be
              * discarded), that is not wasted effort because
              * checkpoint doesn't need to write the page again.
-             *
-             * Once the transaction has given up it's snapshot it
-             * is no longer safe to reconcile pages. That happens
-             * prior to the final metadata checkpoint.
-             *
-             * XXX Only attempt this eviction when there are no
-             * readers older than the checkpoint.  Otherwise, a bug
-             * in eviction can mark the page clean and discard
-             * history, causing those reads to incorrectly see
-             * newer versions of data than they should.
              */
             if (!WT_PAGE_IS_INTERNAL(page) && page->read_gen == WT_READGEN_WONT_NEED &&
-              !tried_eviction && F_ISSET(&session->txn, WT_TXN_HAS_SNAPSHOT) &&
-              (!F_ISSET(txn, WT_TXN_HAS_TS_READ) ||
-                txn->read_timestamp == conn->txn_global.pinned_timestamp)) {
+              !tried_eviction) {
```

Comment:
This was the only conflict involved. Someone will need to take a look at this.

Reply:
@agorrod should confirm, but this looks correct to me, that is, the current PR branch looks right.

Reply:
@keithbostic good question about backing out the workaround. This is the change that does that, and it looks correct to me